File: backend/src/main/java/tqs/msev/backend/entity/Reservation.java
package tqs.msev.backend.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@AllArgsConstructor
@NoArgsConstructor
@Builder
@Data
public class Reservation {
    
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @ManyToOne
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @ManyToOne
    @JoinColumn(name = "charger_id", nullable = false)
    private Charger charger;

    @Column(nullable = false)
    private LocalDateTime startTimestamp;

    @Column(nullable = false)
    private LocalDateTime endTimestamp;

    @Column(nullable = false)
    @Builder.Default
    private boolean used = false;

}

File: frontend/src/types/Reservation.ts
import type { Charger } from "./Charger";
import type { User } from "./user";

export type Reservation = {
  id: string;
  user: User;
  charger: Charger;
  startTimestamp: string;
  endTimestamp: string;
  used: boolean;
}

export type createReservation = {
  user: User;
  charger: Charger;
  startTimestamp: string;
  endTimestamp: string;
}